### PR TITLE
New command for adding a linked server

### LINKED ISSUE #6506: Commands for managing Linked Server
We already have a Get command but need the following:

`New-DbaLinkedServer`
`Set-DbaLinkedServer` -- not sure this would even be possible because there is no support for modifying a linked server via T-SQL, e.g. to update the password it has to be dropped and recreated.

ref #6196

### PR DESCRIPTION
**Pull request by lancasteradam**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #6506 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [x] Documentation
 - [ ] Build system
 

### Purpose
 
Basic command to add a new linked server.

### Commands to test

See the integration tests.

### CODE REVIEW

**Comment by andreasjordan** [comment]
So this will be part of #6506?

**Comment by lancasteradam** [comment]
Hi @andreasjordan, I had not seen that issue, but yes I will update the description of this PR and the other one I just added for removing a linked server so that they both reference 6506.

Thanks,
Adam

**Comment by andreasjordan** [comment]
There is also this one: #6196

**Comment by lancasteradam** [comment]
When I tried the "set" workflow SMO displayed an error message indicating that the linked server has to be dropped and recreated.

**Comment by andreasjordan** [comment]
Ok, then maybe you can add the messages to the feature request and close it.

**Comment by potatoqualitee** [comment]
Looks nice! Any chance you can add a LoginCredential or something similarly named and allow them to create a Linked Server login?

image

If it's too much work, we can add it later.

**Comment by lancasteradam** [comment]
@potatoqualitee, would it be ok to accept the current scope of this PR as a phase 1? I could come back to the login piece at a later date as a phase 2. I would want to proceed very carefully on that part because of the need for secure credential handling.

@andreasjordan, I added the comment to #6196, but I don't have permission to close issues.

File: functions/New-DbaLinkedServer.ps1 [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    if (Test-Bound ProviderString) {
+                        $newLinkedServer.ProviderString = $ProviderString
+                    }
+
+                    if (Test-Bound Catalog) {
+                        $newLinkedServer.Catalog = $Catalog
+                    }
+
+                    $newLinkedServer.Create()
+                    $newLinkedServer
```
Comment (andreasjordan) [inline functions/New-DbaLinkedServer.ps1:137]: Please use Get-DbaLinkedServer to get a nice object with the custom properties for the new linked server:
```
$instance |  Get-DbaLinkedServer -LinkedServer $LinkedServer
```

File: functions/New-DbaLinkedServer.ps1 [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        if (Test-Bound -Not -ParameterName LinkedServer) {
+            Stop-Function -Message "LinkedServer is required"
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            $InputObject += Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+        }
+
+        foreach ($instance in $InputObject) {
```
Comment (andreasjordan) [inline functions/New-DbaLinkedServer.ps1:102]: If something is a server SMO with an active connection, so the output of Connect-DbaInstance, we call this $server. And there are definitly server SMOs in $InputObject so I would change this line to 
```
foreach ($server in $SqlInstance) {
```

**Review by andreasjordan — CHANGES_REQUESTED** [review]

**Comment by lancasteradam** [comment]
Thank you @potatoqualitee and @andreasjordan. I've incorporated the feedback mentioned by @andreasjordan.

**Review by andreasjordan — APPROVED** [review]

**Comment by andreasjordan** [comment]
Approved both pull request.

**Comment by potatoqualitee** [comment]
Fantastic 💯 !

**Pull request merged**